### PR TITLE
Add a step to extract artifacts before SDL run

### PR DESCRIPTION
**Pull request by sunandabalu**
This is needed in order to run BinSkim. 
Arcade Test Build - https://dev.azure.com/dnceng/internal/_build/results?buildId=238907&view=results

### CODE REVIEW

File: eng/common/sdl/extract-artifact-packages.ps1 [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  param( 
+    [string] $PackagePath                                 # Full path to a NuGet package
+  )
+  
+  if (!(Test-Path $PackagePath)) {
+    Write-PipelineTaskError "Input file does not exist: $PackagePath"
+    ExitWithExitCode 1
+  }
+  
+  $RelevantExtensions = @(".dll", ".exe", ".pdb")
```
Comment (JohnTortugo) [inline eng/common/sdl/extract-artifact-packages.ps1]: You really only need these?

File: eng/common/sdl/extract-artifact-packages.ps1 [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  try {
+    $zip = [System.IO.Compression.ZipFile]::OpenRead($PackagePath)
+
+    $zip.Entries | 
+    Where-Object {$RelevantExtensions -contains [System.IO.Path]::GetExtension($_.Name)} |
+      ForEach-Object {
+          $Extension = [System.IO.Path]::GetExtension($_.Name)
+          $FakeName = -Join((New-Guid), $Extension)
+          $TargetFile = Join-Path -Path $ExtractPath -ChildPath $FakeName
```
Comment (JohnTortugo) [inline eng/common/sdl/extract-artifact-packages.ps1]: The file is renamed in this extraction process. Won't you need the name of the file later? For logging or reporting things?

**Comment by sunandabalu** [comment]
> Looks good to me. I'd like to understand better what do you need though. What are you going to do with the extracted files?
> 
> BTW: your test build failed. Ping me if you need help with that.

The extracted files are needed for BinSkim, which runs only against .dlls, .exes and .pdbs.

File: eng/common/sdl/extract-artifact-packages.ps1 (hunk 1 above)
Comment (sunandabalu) [reply eng/common/sdl/extract-artifact-packages.ps1]: Yes for BinSkim, which runs only against .dlls, .exes and .pdbs.

File: eng/common/sdl/extract-artifact-packages.ps1 (hunk 2 above)
Comment (sunandabalu) [reply eng/common/sdl/extract-artifact-packages.ps1]: Oops, yes I do need the real name. I'll fix that.

**Comment by sunandabalu** [comment]
Arcade Test Build - https://dev.azure.com/dnceng/internal/_build/results?buildId=240037&view=results

**Pull request merged**